tools/windows-node-installer/pkg/cloudprovider/factory.go

```diff
@@ -25,6 +54,25 @@ func CloudProviderFactory(kubeconfigPath, credentialPath, credentialAccountID, r
 	}
 
 	switch provider := cloudProvider.Type; provider {
+	case "Azure":
+		//Gather the azure related information.
+		//Read the .auth file to get the subscrptionID and other details.
+		data, err := readJSON("azureauth_location")
+		template, err := readJSON("windowsnode-template.json")
+		if err != nil {
+			log.Fatal("Unable to read template file")
+			return nil, err
+		}
+		params, err := readJSON("windowsnode-params.json")
+		if err != nil {
+			log.Fatal("Unable to read parameters file")
+			return nil, err
+		}
+		azureinfo := svc{resourceGroupName: cloudProvider.Azure.ResourceGroupName,
+			resourceGroupLocation: "centralus", deploymentName: "windowsNode",
```

Comment:
Remove hardcodings

Reply:
These should be inputs that the user provides. Is the `deploymentName` the name of the VM? If yes please match whatever naming convention the installer is following for Linux nodes. Just add `windows` to it. Example: `cluster-name-abc12-windows-worker-us-east-1a-cnpfl`.

Reply:
make sense.